pkg/controller/job/indexed_job_utils.go

```diff
@@ -201,6 +201,12 @@ func firstPendingIndexes(jobCtx *syncJobCtx, count, completions int) []int {
 			active.Insert(ix)
 		}
 	}
+	for _, p := range jobCtx.terminatingPods {
```

Comment:
@mimowo not entirely sure on this one.  

I see that for IndexedJobs we are iterating over `firstPendingIndexes` and we use that to determine creation.  

So in this case, I add terminating to a list.

Reply:
I think this is correct, we don't want to be recreating indexes which yet have terminating pod. Such a pod could still reach `Succeeded` phase and so recreation would not be needed.

Reply:
Do we need `terminatingPods` in any other place? Maybe we could just compute it here, and remove from jobCtx? Also, I prefer to wrap the code in feature gate checks to make it clear to readers that the code does not run by default.

Reply:
removed.